shesha-reactjs/src/components/buttonGroupConfigurator/buttonGroupItem.tsx

```diff
@@ -34,7 +34,7 @@ export const ButtonGroupItem: FC<IButtonGroupItemProps> = ({ item, actualModelCo
   const { styles } = useStyles();
   const actualItem = useMemo(() => getActualModel({ ...item, actionConfiguration }, actualModelContext)
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    , [item.label, item.icon, item.tooltip, item.name, actualModelContext]);
+    , [{ ...item }, actualModelContext]);

```

Comment:
You must not do that, your implementation just always creates a new object and the recalculation will be performed at each rendering regardless of changes in properties

Reply:
When adding item as a dependency as it is, changes did not reflect the applied styles immediately since , till I started listing the style properties, which seemed a bit excessive because there were many properties to list, but now that you explain it, I see that my approach is not ideal and can affect performance.